app/controllers/collections_topics_controller.rb

```diff
@@ -0,0 +1,14 @@
+class CollectionsTopicsController < ApplicationController
+
+  def create
+    @topic = Topic.find(params[:topic_id])
+    @collection = Collection.find(params[:add_topic_to_collection])
+
+    if @topic.update(collections_topics_attributes: [{collection_id: @collection.id, topic_id: @topic.id}])
```

Comment:
Well done for getting this to work, but this isn't correct. You are only allowing users to assign a topic to a single collection. If that's what you wanted, then we went way overboard on the many to many association.
I'm pretty sure the associations are are correct for the desired user functionality, and it is the controller and form functionality that is too simplistic.

Reply:
I was going to add the ability to add the same topic to multiple collections later... you _can_ add topics to multiple collections this way, just manually one at a time... I'll make the changes now though, I've been looking at checkbox forms for editing collections so I think 🤞 I should be able to do it (well, get it to 'werk' anyway)

Reply:
Ah, so without the `_delete` attributes it just creates new `collections_topics`, so you can do it multiple times to add multiple topics (?)

Even if that's the case, there's no way to see which collections a topic belongs to. The advantage of checkboxes or multi-selects is that it shows which collections have already been selected. At present you could end up accidentally trying to add the topic to a collection that it already belongs to, and, with the database indexes as they are, that would blow up with an exception, I think.

Reply:
Yeah exactly, I just need to do it multiple times... you know it works as when you add one you are redirected to the collection show page where the topic you have just added will be listed below the collection name... that said, I do think your way with checkboxes is a better user experience... am in the process of changing the form to a checkbox one 👍